[MM-42416] Calls: RHS Custom post styling

cpoile opened the pull request:
#### Summary
- It looks like the new avatar styling affected the enclosing custom post styling. This fixes it. 

- How the custom post /used/ to look (before the avatar styling):
<img width="546" alt="image" src="https://user-images.githubusercontent.com/1490756/158474547-f14da536-ada3-43e8-ba67-2070750905be.png"> <img width="420" alt="image" src="https://user-images.githubusercontent.com/1490756/158474557-0fe1cf8e-6358-4d46-bcb2-eb47506e79ae.png">


- How the custom post looks on main (after the avatar styling):
<img width="480" alt="image" src="https://user-images.githubusercontent.com/1490756/158474794-df122dda-240f-4733-a645-110865924f08.png"> <img width="410" alt="image" src="https://user-images.githubusercontent.com/1490756/158474816-fc0f8687-d098-4386-84b1-dcaa7cb870e0.png">


- How it looks now:
<img width="473" alt="image" src="https://user-images.githubusercontent.com/1490756/158474981-4e52227c-334d-4dc3-a09e-912c300bf15f.png"> <img width="417" alt="image" src="https://user-images.githubusercontent.com/1490756/158474988-2c69e129-ded1-4377-8446-30f01a116eed.png">

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-42416

abhijit-singh reviewed (commented):
Thanks for fixing this @cpoile. Difficult to test it, but would the layout break when there are a couple more avatars to show over there? Here's a proposed layout I had in the designs if it's not too complicated to implement: 
<img width="356" alt="image" src="https://user-images.githubusercontent.com/14369607/158585205-16b59fc4-bc96-40de-b5d2-6c9ed99b20fe.png">

cpoile commented:
@abhijit-singh I've tried to match the figmas, but I couldn't figure out a simple way to conditionally add the separator (either as a div or a border) based on whether the line wrapped. According to what I found online, it's not possible to detect whether a wrap occurred and apply css conditionally, unless you get into using JS (which I think is a bit of overkill for this). I also converted the hardcoded colors to theme colors.

@streamer45 I converted this component to styled-components -- once I did that, the rest was much easier.

Here are the various combinations and some theme versions:
<img width="736" alt="image" src="https://user-images.githubusercontent.com/1490756/158671252-6b022d3c-77d4-4d5a-8287-1672befe1a9a.png">
<img width="403" alt="image" src="https://user-images.githubusercontent.com/1490756/158671323-14dc42a3-7889-4db4-82fd-0b93e6ee602c.png">

<img width="776" alt="image" src="https://user-images.githubusercontent.com/1490756/158671168-7611206e-c73c-49c6-9a65-1040e4cfab82.png">
<img width="544" alt="image" src="https://user-images.githubusercontent.com/1490756/158671404-b4705cbe-df28-42d1-96f1-90ab0a7f0431.png">
<img width="403" alt="image" src="https://user-images.githubusercontent.com/1490756/158671474-408e10f7-e23b-4272-9299-5b4467cb14b7.png">
<img width="405" alt="image" src="https://user-images.githubusercontent.com/1490756/158671835-1e7676dc-567b-4b29-9eec-6df5207c4ac2.png">
<img width="725" alt="image" src="https://user-images.githubusercontent.com/1490756/158671637-8ee0d37f-2271-4abd-a72a-0865f7a7cd8b.png">
<img width="400" alt="image" src="https://user-images.githubusercontent.com/1490756/158671693-b9364e58-546f-4aad-8ec8-bb677193b859.png">
<img width="399" alt="image" src="https://user-images.githubusercontent.com/1490756/158671978-bff5ad3b-1bca-47d9-aedc-11807b0733a5.png">

streamer45 commented on webapp/src/components/post_type/component.tsx:
```diff
@@ -21,163 +19,182 @@ interface Props {
-
-        return (
-            <div
-                className='call-thread'
```
Could we keep the class? This is actually used by e2e tests to select the element.

cpoile replied:
Oh! Hah, sorry. Will change that to a `data-testid` then (or I'll probably try to delete it in the future when I can't find the css class it belongs to ;) )

cpoile replied:
I can't seem to get the playwright tests to run (they keep stalling), what is the command you use?

streamer45 replied:
That would be `npx playwright test` from within the `e2e` directory. Just bear in mind I was getting some failures earlier on which are unrelated to this PR. I'd suggest to only test the relevant test by temporarily adding `.only`, here.:

https://github.com/mattermost/mattermost-plugin-calls/blob/f1a38f10c9b9523ab3150bb9ddc773b7ac9b6129/e2e/tests/join_call.spec.ts#L54


```js
test.only('call thread', async ({page}) => {  ...
```

streamer45 replied:
Also that assumes MM is running at `http://localhost:8065` with the plugin deployed and enabled. You can pass a custom URL with the `MM_SITE_URL=` env var.

cpoile commented on webapp/.eslintrc.json:
```diff
@@ -48,7 +48,7 @@
     "arrow-body-style": 0,
     "arrow-parens": [
-      2,
+      1,
```
this (and below) makes the prop arrow functions so much easier to read.

streamer45 commented on webapp/src/components/connected_profiles.tsx:
```diff
@@ -17,11 +17,11 @@ interface Props {
-    const margin = -8;
     const els = profiles.map((profile, idx) => {
-        const off = diff > 0 ? (profiles.length - idx) * 8 : (profiles.length - 1 - idx) * 8;
+        const rightShift = 8 * idx;
```
I am thinking this is going to regress the spaning issue we've seen in https://github.com/mattermost/mattermost-plugin-calls/pull/11 :thinking:

cpoile replied:
Ah, dammit. Well, we have to pick what we want then -- either:
1. The avatars shift slightly to the left horizontally in the single line post:
<img width="613" alt="image" src="https://user-images.githubusercontent.com/1490756/158677676-f54284af-8f2c-4998-900b-02fd3e2e724c.png">
vs
<img width="610" alt="image" src="https://user-images.githubusercontent.com/1490756/158677720-1c07b054-576f-490d-82eb-be4fb1cf548b.png">
vs
<img width="609" alt="image" src="https://user-images.githubusercontent.com/1490756/158677766-fd7265d6-9fa0-4cdd-98cc-11f2930e5fd2.png">

or:
2. The avatars shift slightly to the right in the double line post (they line up:
<img width="314" alt="image" src="https://user-images.githubusercontent.com/1490756/158677974-7c4d683b-461a-4dcf-b2bf-80d115df9799.png">
<img width="311" alt="image" src="https://user-images.githubusercontent.com/1490756/158678084-92b50fa7-675d-46b6-bd53-688fc40a05ba.png">
<img width="310" alt="image" src="https://user-images.githubusercontent.com/1490756/158678150-08082e9c-abb5-4cb8-9efe-65fdbfc2791f.png">

@abhijit-singh your choice!

cpoile replied:
I think I prefer the horizontal shift (1), since it's easier to see the misalignment in (2).

streamer45 replied:
Yeah, with the current logic I don't think we can easily cover both cases. I agree that the vertical misalignment is far more noticeable than an extra margin.

Of course I am still hoping that there's an easier (and proper) way to do this kind of positioning and I just don't know how :p

streamer45 replied:
Brilliant, thanks @cpoile :tada:

streamer45 reviewed (commented):
Great work @cpoile  :+1:   Left some minor comments.
